app/src/main/java/com/cornellappdev/uplift/data/auth/TokenAuthenticator.kt

```
@@ -0,0 +1,65 @@
+                        tokenManager.saveTokens(newAccessToken, refreshToken)
+
+                        // Retry the original request with the new Access Token
+                        response.request.newBuilder()
+                            .header("Authorization", "Bearer $newAccessToken")
+                            .build()
+                    } else {
+                        // Refresh failed (e.g., refresh token expired on backend)
+                        sessionManager.logout()
+                        null
+                    }
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)